aiohttp/web_urldispatcher.py

```diff
@@ -457,6 +478,10 @@ def __init__(self, prefix, directory, *, name=None,
                         'HEAD': ResourceRoute('HEAD', self._handle, self,
                                               expect_handler=expect_handler)}
 
+    @property
+    def canonical(self):
+        return self._prefix + str(self._directory)
```

Comment:
Looks like there is an error here.
Canonical URL should contain the prefix only, the directory is a *local* path.
It never returns in URLs but `url_for()` constructs a return value from the prefix and a *relative* to the directory path.

Reply:
Ok, I will remove it from StaticResource, as canonical is already the prefix in PrefixResource and StaticResource will inherit the property.